File: app/src/main/java/com/example/kotlin/first/dsl/DslActivity.java
package com.example.kotlin.first.dsl;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.example.kotlin.R;
import com.example.kotlin.first.dsl.test1.DslActivity1;
import com.example.kotlin.first.dsl.test2.DslActivity2;
import com.example.kotlin.first.jetpack.viewmodel.ViewModelActivity;

/**
 * 使用DSL构建专有的语法结构
 */
public class DslActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_first_dsl);
    }

    /**
     * 模拟dependencies
     * @param v
     */
    public void onTest1(View v) {
        startActivity(new Intent(this, DslActivity1.class));
    }

    /**
     * Html 标签
     * @param v
     */
    public void onTest2(View v) {
        startActivity(new Intent(this, DslActivity2.class));
    }

}

File: app/src/main/java/com/example/kotlin/xiangxue/dsl/DslActivity.kt
package com.example.kotlin.xiangxue.dsl

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.kotlin.R

/**
 * Kotlin DSL
 */
class DslActivity : AppCompatActivity() {

    private var tvTitle : TextView? = null
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_dsl)
        tvTitle = findViewById(R.id.tv_title)
        tvTitle?.text = "Kotlin DSL(领域特定语言)"
    }

    /**
     * DSL简单例子
     */
    fun onDsl1(v : View) {
        startActivity(Intent(this, DslActivity1::class.java))
    }

    /**
     * 模拟AndroidManifest.xml的解析
     */
    fun onDsl2(v : View) {
        startActivity(Intent(this, DslActivity2::class.java))
    }

    /**
     * DSL简单例子泛型实现
     */
    fun onDsl3(v : View) {
        startActivity(Intent(this, DslActivity3::class.java))
    }

    /**
     * DSL简单例子泛型实现
     */
    fun onDsl4(v : View) {
        startActivity(Intent(this, DslActivity4::class.java))
    }

    /**
     * DSL实现Html语法规则
     */
    fun onHtml(v : View) {
        startActivity(Intent(this, DslHtmlActivity5::class.java))
    }
}

